### ISSUE TITLE
Review prioritization of search suggestions 

### ISSUE THREAD

**Issue by topotropic** [mozilla-mobile/fenix#16391]
Based on https://github.com/mozilla-mobile/fenix/issues/12622

When I enter an already open URL the "Switch to tab"  option is last in the search suggestions list and this list can be long. Is this the correct prioritization based on our latest discussions? 

cc @brampitoyo, @violasong

**Comment by brampitoyo** [comment]
@topotropic I agree that our current prioritisation is incorrect, and must be improved!

How to improve it, exactly?

On https://github.com/mozilla-mobile/fenix/issues/13912#issuecomment-675827395, I posted a simplified table of search suggestions ordering between bookmarks, open tabs and synced tabs.

I know that mobile search suggestions should appear roughly in this order, from top to bottom:

1. Search keywords
   - 1 verbatim – exactly as you typed
   - 3–4 from your chosen search provider
2. Bookmarks, open tabs and synced tabs
   - See table above for simplified ordering
3. History items

But the exact prioritisation formula is probably more complicated than this simple list.

We don’t currently know how things work on other devices (Fennec, iOS, Desktop), but we do know that we want them all to match.

(With only one exception: on Fenix and iOS, search keywords must appear first – because people tend to use mobiles for more search oriented tasks).

Because we knew we wanted to match other platforms, we opened #13955 – an Engineering research issue that will help us calculate our current formula on Fenix vs. Fennec vs. iOS vs. desktop. With a formula on hand, we will know how to apply it to improve Fenix’s search suggestions ordering.

Hope this context helps! It’s something that we need to do sooner rather than later.
